dashboard/serializers.py

```diff
@@ -13,67 +13,60 @@ class UserProgramSearchSerializer:
     @classmethod
     def serialize_enrollments(cls, mmtrack):
         """
-        Serializes a user's enrollment data for search results
+        Serializes a user's enrollment data for search results in such a way that enrollments
+        in multiple runs of a single course will result in just one serialization.
 
         Args:
             mmtrack (MMTrack): An MMTrack object
         Returns:
-            list: Serialized courses
+            list: Serialized course enrollments
         """
-        enrollment_status_map = {}
-        course_runs = mmtrack.get_all_enrolled_course_runs()
-        for course_run in course_runs:
+        serialized_enrollments_map = {}
+        for course_run in mmtrack.get_all_enrolled_course_runs():
             course_title = course_run.course.title
-            is_verified = mmtrack.is_enrolled_mmtrack(course_run.edx_course_key)
-            # If any course run for this course was verified/paid, maintain the verified status
-            enrollment_status_map[course_title] = enrollment_status_map.get(course_title) or is_verified
-        serialized_enrollments = []
-        for course_title, is_verified in enrollment_status_map.items():
-            enrollment_status = 'Paid' if is_verified else 'Auditing'
-            serialized_enrollments.extend([
-                {'level': 1, 'value': course_title, 'ancestors': []},
-                {'level': 2, 'value': enrollment_status, 'ancestors': [course_title]}
-            ])
-        return serialized_enrollments
+            # If any course run for this course was verified/paid, count it as verified
+            if course_title not in serialized_enrollments_map or \
+                    serialized_enrollments_map[course_title]['payment_status'] == 'Auditing':
+                serialized_enrollments_map[course_title] = cls.serialize_enrollment(mmtrack, course_run)
+        return list(serialized_enrollments_map.values())
 
     @classmethod
-    def serialize_final_grades(cls, mmtrack):
+    def serialize_enrollment(cls, mmtrack, course_run):
         """
-        Serializes a user's final grades
+        Serializes information about a user's enrollment in a course run
 
         Args:
             mmtrack (MMTrack): An MMTrack object
+            course_run (CourseRun): An enrolled CourseRun
         Returns:
-            list: Serialized final grades
+            dict: Serialized course enrollment
         """
-
-        serialized_final_grades = []
-        final_grades = mmtrack.get_all_final_grades()
-        for final_grade in final_grades.values():
-            serialized_final_grades.append({
-                'title': final_grade.course_run.course.title,
-                'grade': final_grade.grade_percent
-            })
-        return serialized_final_grades
+        course_title = course_run.course.title
+        is_verified = mmtrack.is_enrolled_mmtrack(course_run.edx_course_key)
```

Comment:
Since we are indexing enrollments in courses in cases where users have a final grade but are missing CachedErollment, `is_enrolled_mmtrack` returns false for users that have only final grade. We might want to use 
`is_verified = mmtack.has_paid(course_run.edx_course_key)`.